package com.app.daoimpl;

import java.sql.Connection;
import java.sql.PreparedStatement;
import java.sql.SQLException;

import org.apache.log4j.Logger;

import com.app.dao.UsersDAO;
import com.app.dao.dbutil.MySqlDbConnection;
import com.app.exception.BusineesException;
import com.app.model.User;

public class UsersDAOImpl implements UsersDAO {
	private static final Logger log =Logger.getLogger(UsersDAOImpl.class);
	

	@Override
	public int createUser(User user) throws BusineesException {
	int createUser=0;
	try(Connection connection =MySqlDbConnection.getConnection()){
		String sql="insert into users (firstName,lastName,email,password) values(?,?,?,? )";
		PreparedStatement preparedStatement=connection.prepareStatement(sql);
		preparedStatement.setString(1, user.getFirstName());
		preparedStatement.setString(2, user.getLastName());
		preparedStatement.setString(3, user.getEmail());
		preparedStatement.setString(4, user.getPassword());
		
		createUser=preparedStatement.executeUpdate();
	} catch (ClassNotFoundException | SQLException e) {
	   log.error(e);
	   throw new BusineesException("Internal Error Occured contact SysAdmin");
	}
	
		return createUser;
	}
	
	

}
